game-core/src/test/java/games/strategy/util/PointFileReaderWriterTest.java

```diff
@@ -107,6 +160,103 @@ public void shouldReadMultiplePointsPerName() throws Exception {
           "54 Sea Zone", Arrays.asList(new Point(2011, 2021), new Point(2012, 2022)),
           "Philippines", Arrays.asList(new Point(3011, 3021)))));
     }
+
+    @Test
+    public void shouldErrorOnDuplicateKey() {
+      final String content1 = ""
+          + "54 Sea Zone  (1011,1021)\n"
+          + "54 Sea Zone  (1011,1021)";
+
+      final Exception e1 = assertThrows(IOException.class,
+          () -> readFromString(PointFileReaderWriter::readOneToMany, content1));
+      assertTrue(e1.getMessage().contains("54 Sea Zone"));
+
+
+      final String content2 = ""
+          + "54 Sea Zone  (1011,1021)\n"
+          + "54 Sea Zone  (      1011  , 1021    )   ";
+
+      final Exception e2 = assertThrows(IOException.class,
+          () -> readFromString(PointFileReaderWriter::readOneToMany, content2));
+      assertTrue(e2.getMessage().contains("54 Sea Zone"));
+
+
+      final String content3 = ""
+          + "54 Sea Zone  (1011,1021)  (1012,1022)  (1013,1023)\n"
+          + "54 Sea Zone  (2011,2021)  (2012,2022)";
+
+      final Exception e3 = assertThrows(IOException.class,
+          () -> readFromString(PointFileReaderWriter::readOneToMany, content3));
+      assertTrue(e3.getMessage().contains("54 Sea Zone"));
+    }
+  }
+
+  @Nested
+  public final class ReadOneToManyPlacementsTest {
+    @Test
+    public void shouldNotCloseStream() throws Exception {
+      final InputStream is = spy(new ByteArrayInputStream(new byte[0]));
+
+      PointFileReaderWriter.readOneToManyPlacements(is);
+
+      verify(is, never()).close();
+    }
+
+    @Test
+    public void shouldReturnEmptyMapWhenStreamIsEmpty() throws Exception {
+      assertThat(IoUtils.readFromMemory(new byte[0], PointFileReaderWriter::readOneToManyPlacements),
+          is(Collections.emptyMap()));
+    }
+
+    @Test
+    public void shouldReadMultiplePlacementsPerName() throws Exception {
```

Comment:
@RoiEXLab The tests here look good to me. Essentially the only time "overflowToLeft" should be true is if the line ends with " | overflowToLeft=true". Any other cases, it should be false. We also control the writing of that text in the map creator so as long as the 2 match up, we shouldn't have any issues.